contracts/locator/src/contract.rs

```diff
@@ -0,0 +1,147 @@
+        QueryMsg::Ownership {} => Ok(to_json_binary(&get_ownership(deps.storage)?)?),
+        QueryMsg::FactoryInstance { name } => query_factory_instance(deps, name),
+        QueryMsg::FactoryInstances {} => query_factory_instances(deps),
+    }
+}
+
+pub fn query_factory_instance(deps: Deps<NeutronQuery>, name: String) -> StdResult<Binary> {
+    let factory_addr = STATE.load(deps.storage, name)?.factory_addr;
+    to_json_binary(&FactoryInstance {
+        addr: factory_addr.to_string(),
+        contracts: deps
+            .querier
+            .query_wasm_smart(factory_addr.clone(), &FactoryQueryMsg::State {})?,
+    })
+}
+
+pub fn query_factory_instances(deps: Deps<NeutronQuery>) -> StdResult<Binary> {
+    let drop_instances: Vec<FactoryInstance> = STATE
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|item| {
+            item.map(|(_key, value)| FactoryInstance {
+                addr: value.factory_addr.clone(),
+                contracts: deps
+                    .querier
+                    .query_wasm_smart(value.factory_addr.clone(), &FactoryQueryMsg::State {})
+                    .unwrap(),
+            })
+            .unwrap()
+        })
+        .collect();
+    to_json_binary(&drop_instances)
+}
+
+pub fn query_chain(deps: Deps<NeutronQuery>, name: String) -> StdResult<Binary> {
+    let chain = STATE.load(deps.storage, name.clone())?;
+    to_json_binary(&DropInstance {
+        name,
+        factory_addr: chain.factory_addr,
+    })
+}
+
+pub fn query_chains(deps: Deps<NeutronQuery>) -> StdResult<Binary> {
+    let drop_instances: StdResult<Vec<_>> = STATE
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|item| {
+            item.map(|(key, value)| DropInstance {
+                name: key,
+                factory_addr: value.factory_addr,
+            })
+        })
+        .collect();
+    let drop_instances = drop_instances?;
+    to_json_binary(&drop_instances)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> ContractResult<Response<NeutronMsg>> {
+    match msg {
+        ExecuteMsg::AddChains { chains } => execute_add_chains(deps, env, info, chains),
+        ExecuteMsg::RemoveChains { names } => execute_remove_chains(deps, env, info, names),
+        ExecuteMsg::UpdateOwnership(action) => {
+            update_ownership(deps.into_empty(), &env.block, &info.sender, action)?;
+            Ok(Response::new())
+        }
+    }
+}
+
+pub fn execute_remove_chains(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: Vec<String>,
+) -> ContractResult<Response<NeutronMsg>> {
+    cw_ownable::assert_owner(deps.storage, &info.sender)?;
+    let mut attrs: Vec<Attribute> = Vec::new();
+    msg.iter().for_each(|name| {
+        STATE.remove(deps.storage, name.to_string());
+        attrs.push(attr("remove", name))
```

Comment:
not sure it will properly add attribute with same name but with different value.
You can use `STATE.clear(deps.storage)` instead, and something like `attrs.push(attr("removed_chains", names.join(",")));`

Reply:
`STATE.clear(deps.storage)` will remove entire data, won't it?

Reply:
Yes. Just saw that you wanted to remove only specific chains. So feel free to keep it. But I'd recommend you to change approach in attrs processing